packages/itwinui-react/src/core/Tile/TileNew.tsx

```diff
@@ -0,0 +1,646 @@
+
+type TileTypeIndicatorOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+// ----------------------------------------------------------------------------
+// Tile.Badge component
+
+const TileBadge = polymorphic.div('iui-tile-thumbnail-badge-container');
+TileBadge.displayName = 'Tile.Badge';
+
+type TileBadgeOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+// ----------------------------------------------------------------------------
+// Tile.Name component
+
+const TileName = React.forwardRef(
+  ({ as: Element = 'div', children, ...rest }, ref) => {
+    const {
+      status,
+      isLoading,
+      isSelected,
+      isNew,
+      isActionable,
+      isDisabled,
+      onClick,
+    } = useSafeContext(TileContext);
+
+    return (
+      <Element className='iui-tile-name' ref={ref} {...rest}>
+        <TitleIcon
+          isLoading={isLoading}
+          isSelected={isSelected}
+          isNew={isNew}
+          status={status}
+        />
+        <span className='iui-tile-name-label'>
+          {isActionable && onClick ? (
+            <LinkAction
+              as='button'
+              onClick={!isDisabled ? onClick : undefined}
+              aria-disabled={isDisabled}
+            >
+              {children}
+            </LinkAction>
+          ) : (
+            children
+          )}
+        </span>
+      </Element>
+    );
+  },
+) as PolymorphicForwardRefComponent<'div', TileNameOwnProps>;
+TileName.displayName = 'TileNew.Name';
+
+type TileNameOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+// ----------------------------------------------------------------------------
+// Tile.ContentArea component
+
+const TileContentArea = polymorphic.div('iui-tile-content');
+TileContentArea.displayName = 'TileNew.ContentArea';
+
+type TileContentAreaOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+// ----------------------------------------------------------------------------
+// Tile.Description component
+
+const TileDescription = polymorphic.div('iui-tile-description');
+TileDescription.displayName = 'TileNew.Description';
+
+type TileDescriptionOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+// ----------------------------------------------------------------------------
+// Tile.Metadata component
+
+const TileMetadata = polymorphic.div('iui-tile-metadata');
+TileMetadata.displayName = 'TileNew.Metadata';
+
+type TileMetadataOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+// ----------------------------------------------------------------------------
+// Tile.MoreOptions component
+
+const TileMoreOptions = React.forwardRef((props, ref) => {
+  const { as: Element = 'div', className, children, ...rest } = props;
+  const [isMenuVisible, setIsMenuVisible] = React.useState(false);
+  const showMenu = React.useCallback(() => setIsMenuVisible(true), []);
```

Comment:
nitpicky but can these calls just immediately go inside of `onShow` and `onHide`, leaving no need to make these separate variables?

Reply:
Yup I moved these functions to inline :d